butane_cli/src/lib.rs

```diff
@@ -259,10 +261,46 @@ pub fn get_migrations() -> Result<FsMigrations> {
     Ok(migrations::from_root(root))
 }
 
+/// The members of a workspace are given a PackageId, which contains the directory
+/// name however it needs to be extracted.
+pub fn extract_package_directory(
```

Comment:
I'm not a big fan of of using `package_id.repr` because it's documented to be unstable. I tried out the following locally and it seems to work. The key point here is that we can use the `PackageId` to match against the list of `Package`. There's a lot more info on `Package` including the path to the project's `Cargo.toml`
```
pub fn extract_package_directory(
    packages: &Vec<cargo_metadata::Package>,
    package_id: cargo_metadata::PackageId,
) -> Result<std::path::PathBuf> {
    let pkg = packages
        .iter()
        .find(|p| p.id == package_id)
        .ok_or(anyhow!("No package found"))?;
    let parent = pkg.manifest_path.parent().unwrap();
    Ok(parent.to_owned().into())
}
```